### ISSUE TITLE
Remove Duplicate Classes

### ISSUE THREAD

**Issue by jayanderson** [signalfx/signalfx-java#59]
We're beginning our integration with signalfx. One rule we have in place for our builds is to avoid duplicate classes in our classpath. This is enforced with the maven enforcer plugin. Here is an example of a class being duplicated:

From the signalfx jar:
```
→ jar -tvf signalfx-codahale-0.0.45.jar | grep com.codahale.metrics.MetricRegistry.class
 14822 Mon Mar 26 14:41:08 MST 2018 com/codahale/metrics/MetricRegistry.class
```

From the dropwizard jar:
```
→ jar -tvf metrics-core-3.1.5.jar | grep com.codahale.metrics.MetricRegistry.class
 12076 Fri Jun 02 19:59:20 MST 2017 com/codahale/metrics/MetricRegistry.class
```

Is it possible to remove these duplicate classes and rely on maven dependencies instead? Either that or namespace them under a different package to avoid this conflict?

**Comment by mpetazzoni** [comment]
Most people that use `signalfx-codahale` don't have a distinct dependency on `metrics-core`. But we could change that to mark this dependency as compile-time only in a future release. It would force people to define an explicit, separate dependency on `metrics-core` though -- and hopefully of the right version.

@tedoc2000 Any thoughts on this?

**Comment by jayanderson** [comment]
We're migrating existing services which use the `MetricRegistry` class for recording metrics so it is an explicit dependency. The main thing we need out of this signalfx jar is the `SignalFxReporter` class to report our metrics.

**Comment by mpetazzoni** [comment]
Makes sense. I'll release a new version today that fixes the problem. Hang tight.

**Comment by cahudson94** [comment]
Worked pretty great resolved the issues around the classes there is still overlap with jetty-dir as we have a dependency on jetty-util and a new bit of overlap from apach httpclient and mozilla/public-suffix-list.txt. When I went to make exceptions for these through duplicate finder plugin it threw up a bunch of new class conflicts fro some reason coming from googles findbugs which is brought in by the parent pom but not needed for the module using sfx. Got it working these are mostly some side notes.

**Comment by jayanderson** [comment]
Perfect. Thanks for the quick work!
